kombu/utils/limits.py

```diff
@@ -23,32 +22,21 @@ class TokenBucket:
         section of any multithreaded code.
     """
 
-    #: The rate in tokens/second that the bucket will be refilled.
-    fill_rate = None
+    def __init__(
+        self,
+        fill_rate: str | int | float,
+        capacity: int = 1
```

Comment:
I guess `capacity` follows the same type as `fill_rate`?

Reply:
My understanding is that capacity is a discrete measure, and should be an int. At the same time, it's used like this in the class (unchanged code):

```python
self._tokens = min(self.capacity, self._tokens + delta)
```

Since mixing ints and floats causes mypy errors, I decided to cast `self.capacity` and `self._tokens` to floats in the init instead of ignoring the mypy error. Don't think they are used in a manner that require them to be ints anywhere else.

Does that make sense?